packages/matchbox/src/components/Radio/Group.js

```diff
@@ -1,19 +1,50 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { Label } from '../Label';
-import styles from './Radio.module.scss';
+import { Box } from '../Box';
+import { ScreenReaderOnly } from '../ScreenReaderOnly';
+import { createPropTypes } from '@styled-system/prop-types';
+import styled from 'styled-components';
+import { margin } from 'styled-system';
+import { pick } from '@styled-system/props';
 
-const Group = ({ children, label, required }) => (
-  <div className={styles.Group}>
-    {label && <Label className={styles.GroupLabel}>{label}{required && ' *'}</Label>}
-    {children}
-  </div>
-);
+const StyledGroup = styled('fieldset')`
+  border: none;
+  padding: 0;
+  ${margin}
+`;
+
+function Group(props) {
+  const { children, label, labelHidden, required, ...rest } = props;
+  const systemProps = pick(rest);
+
+  return (
+    <StyledGroup {...systemProps}>
+      {label && !labelHidden && (
+        <Box as="legend" lineHeight="200" fontSize="200" mb="100">
+          {label}
+          {required && (
+            <Box as="span" pr="200" aria-hidden="true">
+              *
+            </Box>
+          )}
+        </Box>
+      )}
+      {label && labelHidden && (
+        <ScreenReaderOnly>
+          <legend>{label}</legend>
+        </ScreenReaderOnly>
+      )}
+      {children}
+    </StyledGroup>
+  );
+}
 
 Group.propTypes = {
   children: PropTypes.node.isRequired,
   label: PropTypes.node,
```

Comment:
This one should be `required` if possible

Reply:
And as a result, the additional checks for `label` shouldn't be required in the JSX